### DISCUSSION TITLE
Conditional styles In JIT-Mode are not generated

### DISCUSSION THREAD

**Discussion by BartBBM** [tailwindlabs/tailwindcss#5647] [Help] [answered]
Hello,
first of all, it works if I don't use `mode: 'jit'`. I'm using the last version of tailwind: `tailwindcss": "^2.2.16`. I'm using `npm run dev (svelte-kit dev)`.
I'm using a div:

`<div class="p-4 m-4 border-2 rounded-lg w-96 border-{color}-200 bg-{color}-100">`

This div is dependent from the variable `color`, which either evaluates to `'green'` or `'red'`. But when in JIT-Mode I can see in the dev-tools of chrome, that the class name gets generated successfully, but there is not any style related to it.
So it evaluates to `bg-green-100`, but this class is not recognized by jit to get generated. When use hot reload and change the `bg-{color}-100` to `bg-green-100` hardcoded, it obviously works. And when I change back (and do not stop the dev server) it continuously works (only for the hardcoded color).
I think it's because the style gets loaded into cache and thus has not to be newly generated.
Aren't you supposed to use this type of conditional style or is this a bug?
I would be pleased if somebody could help me :)
Thanks in advance
[arguments.zip](https://github.com/tailwindlabs/tailwindcss/files/7255217/arguments.zip)

**Comment by adamwathan** [comment, marked as answer]
Hey! You still need to write “purgeable” markup with the JIT engine, this should explain it:

https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html

**Reply by BartBBM** [reply]
Thanks! for answering such a stupid question :)
Love using tailwind ❤️, have a great day!
